### ISSUE TITLE
No automatic deceleration at deceleration point as before

### ISSUE THREAD

**Issue by phildeg31** [flybywiresim/a32nx#4052]
**Mod Version**
"built": "2021-03-18T05:12:53+00:00",
  "ref": "fbw",

**Describe the bug**
Before, when the deceleration point was reached, the target speed taken into account in managed mode was the final approach speed. With the last version of the FBW mod, there is no more automatic change for speed at deceleration point. I have to decelerate manually and to activate the approach phase before landing to have the good behavior.

**To Reproduce**
Do a flight and see if, with speed and heading in managed mode and with only target altitude for approach set manually, the target speed changes when the deceleration point is reached.

**Expected behavior**
Target speed should decrease at deleration point to final approach speed.

**Actual behavior**
Speed does not change at deceleration point (not taken into account?).

**Comment by MisterChocker** [comment]
Do you have pictures or a video when the issue occurred?

**Comment by phildeg31** [comment]
Before deleration point:
![3](https://user-images.githubusercontent.com/74733393/112010884-0132ef00-8b28-11eb-85dc-9e7332af6405.JPG)
On deceleration point:
![4](https://user-images.githubusercontent.com/74733393/112010930-0d1eb100-8b28-11eb-9b1d-bf3ba0909c42.JPG)
After deceleration point:
![5](https://user-images.githubusercontent.com/74733393/112010991-1c056380-8b28-11eb-8063-1735ef8b04e2.JPG)

I have to activate approach phase manually:
![6](https://user-images.githubusercontent.com/74733393/112011034-27f12580-8b28-11eb-8fc1-979ce171db29.JPG)

Before, when deceleration point was reached, then the target speed changed automatically.

**Comment by MisterChocker** [comment]
Yes, this happens because you are in Cruise Flight Phase.

**Comment by phildeg31** [comment]
Yes but why am I consider in this phase?

When deceleration point is reached, then the plane should target VAPP:

  | DeceleratePoint (M) | NAV/ARCPLAN | Indicates where the airplane will start an automatic deceleration toward VAPP (switches into the APP phase).
-- | -- | -- | --

**Comment by MisterChocker** [comment]
Yes, therefore you MUST be in DES phase.

**Comment by phildeg31** [comment]
I am in descent to 3000 ft on the screen (so the target altitude for approach).

**Comment by MisterChocker** [comment]
You are still in cruise flight phase, look at the performance page.

**Comment by phildeg31** [comment]
Yes I know but it is not normal in my case because I started the descent. How do you change to descent phase in your case?

**Comment by MisterChocker** [comment]
Currently you need to initiate the descent via pulling/pushing the alt knob on the fcu. In the future we will also support V/S initiation.

**Comment by phildeg31** [comment]
The altitude was 120000 before starting the descent. After that, when I had the symbol for the descent, I changed to the target altitude (3000ft).

**Comment by MisterChocker** [comment]
Alright, then your Logitech device is not sending the proper events.

**Comment by phildeg31** [comment]
I did not change anything to my panel and its event management (I use Spad.next and configured it months ago) and before it worked. I will try to change in game directly to compare.

**Comment by MisterChocker** [comment]
Yeah, it's not your fault and unfortunately you can't do anything about it. For now you will need to use the ingame fcu.

**Comment by phildeg31** [comment]
But the event seems to be taken into account by the game because the target altitude in game is 3000ft (FL30).

**Comment by MisterChocker** [comment]
No this is due to the recode of the flight phase management system, it hooks into certain events and unfortunately it looks like that  alt knob pull/push seems not to be triggered when you use your modification.

**Comment by phildeg31** [comment]
Do you mean a recode in the mod? If yes, is it possible to hook also this event that worked before please because only one event seems to be given in SPAD.next: 
![image](https://user-images.githubusercontent.com/74733393/112045778-84197100-8b4b-11eb-9dad-61e1f0ed976b.png)

I can try to provide you more information about this event if you explain me how to find it.

**Comment by phildeg31** [comment]
The event used by SPAD.next seems to be used in the code with `SimVar.SetSimVarValue("K:ALTITUDE_SLOT_INDEX_SET", "number", 2);` to set in managed mode. It is the same I think.

**Comment by MisterChocker** [comment]
Do don't know if they are available, the H events I've send earlier are the ones that trigger the code posted by you.
If you are able to edit the event/simvar you are using in SPAD.next to the one I've posted, then everything should work as normal.

**Comment by phildeg31** [comment]
SPAD.next propose sim events (like ALTITUDE_SLOT_INDEX_SET) but I do not lnow if it is possible to send a specific non sim event. It worked fine before so a code must have been deleted making it not working anymore.
If you are available to intercept or to see if there is a simulator event like ALTITUDE_SLOT_INDEX_SET, it could be great to intercept it and do a call the code to change the phase if needed.
I will see what is possible before the end of the week when I have time.

**Comment by MisterChocker** [comment]
> I think it is when you press the button for selected or managed mode no?

yes

**Comment by phildeg31** [comment]
Response for SPAD.next on their Discord: "You can manage LVARs , not (yet) HVARs". So it is not possible to throw a H event for now.

**Comment by MisterChocker** [comment]
Using the simvar updater for this is inappropriate.

Due to the difference between/changes made to the FCU on master vs autopilot, I'm suggesting waiting for integration first and then looking at the issue again to find a good way to solve it.


> By the way, you can update _flapsInMotionSelector() function to not call `SimVar.GetSimVarValue("TRAILING EDGE FLAPS LEFT PERCENT", "percent")` the second time for `this.lastFlapsPosition` because you already have retrieved the value for `const currentFlapsPosition`

Please feel free to contribute to the project.

I am going to close this issue since the reported issue is working as intended.
You can open go ahead and open a new issue for the discovered compatibly issue with SPAD.next.

Joining our discord server may also be a good idea to start discussing the issue and ways to solve it.

**Comment by phildeg31** [comment]
I created this ticket: https://github.com/flybywiresim/a32nx/issues/4101

I would like to participate to the project but I have not enought time because of work and childs taking me a lot of time (I can only play once or twice a month). You (all the team) did a great job.
